### PR TITLE
Pattern Library: Mobile tweaks

### PR DESCRIPTION
**Pull request by fredrikekelund**
Fixes https://github.com/Automattic/dotcom-forge/issues/6278
Fixes https://github.com/Automattic/dotcom-forge/issues/6389

## Proposed Changes

This PR fixes a few small things, primarily on mobile:

- When searching, the `%(count) patterns` heading is now displayed on mobile
- The `Copy, paste, customize` and `All about patterns` sections now use scroll snapping because it's neat
- When searching in all categories, the `All categories` link remains active (this was primarily problematic on mobile, where the dropdown previously didn't have any selected text while searching all categories)

https://github.com/Automattic/wp-calypso/assets/1101677/fba3f4e6-0e46-4c32-97fd-9c87c4d69136

## Testing Instructions

1. Navigate to `/patterns` on mobile. I suggest using the Calypso preview link on your phone.
2. Search for something
3. Ensure that an `N patterns` heading is shown and that the nav dropdown still says `All categories`
4. Navigate to a category
5. Ensure that no `Patterns` heading is shown
6. Go back to `All categories`
7. Scroll down to the `Copy, paste, customize` section and ensure that it now has scroll snapping

### CODE REVIEW

**Comment by fredrikekelund** [comment]
I also took the opportunity to fix the highlight color in the mobile dropdown menu, as mentioned in pdtkmj-2wZ-p2#comment-4711

File: client/my-sites/patterns/components/pattern-library/index.tsx [hunk 1]
```diff
@@ -276,7 +276,11 @@ export const PatternLibrary = ( {
 				{ ! isHomePage && (
 					<PatternLibraryBody className="pattern-library">
 						<div className="pattern-library__header">
-							<h1 className="pattern-library__title">
+							<h1
```
Comment (nightnei) [inline client/my-sites/patterns/components/pattern-library/index.tsx:264]: NIT

```
{ searchTerm ? (
	<h1 className="pattern-library__title pattern-library__title--search">
		{ translate( '%(count)d pattern', '%(count)d patterns', {
			count: patterns.length,
			args: { count: patterns.length },
		} ) }
	</h1>
) : (
	<h1 className="pattern-library__title">
		{ patternTypeFilter === PatternTypeFilter.PAGES &&
			translate( 'Page Layouts', {
				comment: 'Refers to block patterns that contain entire page layouts',
			} ) }
		{ patternTypeFilter === PatternTypeFilter.REGULAR &&
			translate( 'Patterns', {
				comment: 'Refers to block patterns',
			} ) }
	</h1>
) }
```

**Review by nightnei — APPROVED** [review]
LGTM and scroll works cool 👍

File: client/my-sites/patterns/components/pattern-library/index.tsx (hunk 1 above)
Comment (fredrikekelund) [reply client/my-sites/patterns/components/pattern-library/index.tsx:264]: This doesn't seem to reduce indentation, so I'll leave the current code in place

**Pull request merged**